mobile/src/components/courses/courses.tsx

```diff
@@ -49,20 +67,59 @@ const Courses: FC = (): ReactElement => {
   };
 
   const handleSearch = (search: string): void => {
-    dispatch(coursesActions.getCourses({ title: search, categoryKey: '' }));
+    setSearchValue(search);
+  };
+
+  const handleCategorySelect = (id: number): void => {
+    if (activeCategoryId !== id) {
+      setActiveCategoryId(id);
+    } else {
+      setActiveCategoryId(null);
+    }
```

Comment:
It's better to make only one call 
```suggestion
    const newActiveCategoryId = activeCategoryId !== id ? id : null;

    setActiveCategoryId(newActiveCategoryId);
```